piccolo_api/crud/endpoints.py

```diff
@@ -812,6 +813,12 @@ async def post_single(
                 row = await execute_post_hooks(
                     hooks=self._hook_map, hook_type=HookType.pre_save, row=row
                 )
+            try:
+                await AuditLog.post_save_action(
+                    self.table, user_id=request.user.user_id
+                )
+            except AssertionError:
```

Comment:
Where are the `AssertionErrors` being raised?

Reply:
I used this to prevent `AssertionError` from `AuthenticationMiddleware` because we should be able to use `PiccoloCRUD` without session authentication which `AuditLog` depends on (if we use e.g. OAuth in FastAPI docs without this, we get `AssertionError: AuthenticationMiddleware must be installed to access request.user` when trying to save records)

Reply:
I see what you mean. I think the only problem is it ties it really closely to FastAPI's internals, which could change in the future. How about we catch all exceptions and log it? Something like:

```python
try:
     await AuditLog....
except Exception as exception:
    logger.log(exception)
```